ts/utils/fetch.ts

```diff
@@ -114,3 +129,246 @@ export const constantPollingFetch = (
+              customSignValues.forEach(
+                v => (newInit = addHeader(newInit, v.headerName, v.headerValue))
+              );
+              // Prepare custom signature inputs array
+              const customSignatureInputs = customSignValues.map(
+                v => v["signature-input"]
+              );
+              // Prepare custom signature array
+              const customSignatures = customSignValues.map(v => v.signature);
+              // Setup signature array
+              const signatures = [
+                `sig1:${mainSignValue}:`,
+                ...customSignatures
+              ];
+              // Setup signature input array
+              const signatureInputs = [
+                mainSignatureInput,
+                ...customSignatureInputs
+              ];
+              // Add all to their corresponding headers
+              newInit = addHeader(newInit, "signature", signatures.join(","));
+              newInit = addHeader(
+                newInit,
+                "signature-input",
+                signatureInputs.join(",")
+              );
+              return timeoutFetch(input, newInit);
+            });
+          })
+          .catch(function () {
+            return timeoutFetch(input, init);
+          });
+      }
+      return timeoutFetch(input, init);
+    }
+  );
+}
+
+type SignatureConfigForgeInput = {
+  publicKey: PublicKey;
+  keyTag: string;
+  lollipopConfig: LollipopConfig;
+  method: string;
+  inputUrl: URLParse;
+  originalUrl: string;
+};
+
+function forgeSignatureConfig(
+  forgeInput: SignatureConfigForgeInput,
+  keyInfo: KeyInfo,
+  signatureParams: Array<string>
+): SignatureConfig {
+  return {
+    signAlgorithm: getSignAlgorithm(forgeInput.publicKey),
+    signKeyTag: forgeInput.keyTag,
+    signKeyId: keyInfo.publicKeyThumbprint ?? "",
+    nonce: forgeInput.lollipopConfig.nonce,
+    signatureComponents: forgeSignatureComponents(
+      forgeInput.method,
+      forgeInput.inputUrl,
+      forgeInput.originalUrl
+    ),
+    signatureParams
+  };
+}
+
+function extractHttpRequestComponents(input: string, init: RequestInit) {
+  const inputUrl = new URLParse(input, true);
+  const queryString: string | undefined = inputUrl.href.split("?")[1];
+  const method = init.method?.toUpperCase() ?? "";
+  const body = init.body;
+  const bodyString = body as string;
+  const originalUrl =
+    inputUrl.pathname + (queryString ? "?" + queryString : "");
+  return { body, bodyString, inputUrl, queryString, method, originalUrl };
+}
+
+/**
+ * Add a pair header:value to the current fetch init.headers.
+ */
+function addHeader(
+  init: RequestInit,
+  headerName: string,
+  headerValue: string | number
+) {
+  return {
+    ...init,
+    headers: {
+      ...init.headers,
+      [headerName]: headerValue
+    }
+  };
+}
+
+/**
+ * Check if the keyInfo and Request properties are properly initialized for fetching
+ */
+function isKeyInfoAndRquestValid(
```

Comment:
It looks like this method is returning a value (not a boolean), moreover there's a typo

```suggestion
function toRequestAndKeyInfoForLPFetch(
```